Adding `type: "module"` to actiontext caused an error

In [webpack](https://webpack.js.org/guides/ecma-script-modules/) it says

> Imports in ESM are resolved more strictly. Relative requests must include a filename and file extension.

By adding `type: "module"` to [actiontext](https://github.com/rails/rails/blob/main/actiontext/package.json#L6), but not importing with an extension [here](https://github.com/rails/rails/blob/main/actiontext/app/javascript/actiontext/attachment_upload.js#L1), importing actiontext generates an error... I think.

### Steps to reproduce

In my app/javascript/packs/application.js, importing `import "@rails/actiontext"`  generates an error message when running `bin/webpack` or `bin/webpack-dev-server`

It errors with just actiontext, as well an combination of the following:

```
import "core-js/stable";
import "regenerator-runtime/runtime";
import "@rails/activestorage";
import "@rails/actiontext";
import "@rails/ujs";

Rails.start();
```

Using actiontext (7.0.0) with webpacker (6.0.0-rc.6). I tried to create a test repo but could not get `rails webpacker:install` or `rails new --webpacker` (I think maybe it's gone in favor of the new stuff) to run.

### Expected behavior

It should import without an error message

### Actual behavior

It generates the following error

> ERROR in ./node_modules/@rails/actiontext/app/javascript/actiontext/index.js 1:0-54
> Module not found: Error: Can't resolve './attachment_upload' in '/Users/tomify/Desktop/development/visible-voices/node_modules/@rails/actiontext/app/javascript/actiontext'
> Did you mean 'attachment_upload.js'?
> BREAKING CHANGE: The request './attachment_upload' failed to resolve only because it was resolved as fully specified
> (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
> The extension in the request is mandatory for it to be fully specified.
> Add the extension to the request.
> resolve './attachment_upload' in '/Users/tomify/Desktop/development/visible-voices/node_modules/@rails/actiontext/app/javascript/actiontext'
>   using description file: /Users/tomify/Desktop/development/visible-voices/node_modules/@rails/actiontext/package.json (relative path: ./app/javascript/actiontext)
>     Field 'browser' doesn't contain a valid alias configuration
>     using description file: /Users/tomify/Desktop/development/visible-voices/node_modules/@rails/actiontext/package.json (relative path: ./app/javascript/actiontext/attachment_upload)
>       Field 'browser' doesn't contain a valid alias configuration
>       /Users/tomify/Desktop/development/visible-voices/node_modules/@rails/actiontext/app/javascript/actiontext/attachment_upload doesn't exist
> 
> webpack 5.64.4 compiled with 1 error and 1 warning in 23809 ms

### System configuration

**Rails version**: 7.0.0
**Ruby version**: 2.7.1

Here is my package.json with potentially relevant libraries. I tested different combinations as well as removing some of them, but the error persists.

```
{
  "name": "app",
  "browserslist": ["defaults"],
  "dependencies": {
    "@babel/core": "^7.16.0",
    "@babel/plugin-proposal-export-default-from": "^7.16.0",
    "@babel/preset-react": "^7.16.0",
    "@rails/actiontext": "^7.0.0",
    "@rails/activestorage": "^7.0.0",
    "@rails/ujs": "7.0.0",
    "@rails/webpacker": "6.0.0-rc.6",
    "core-js": "^3.18.2",
    "css-loader": "^6.3.0",
    "css-minimizer-webpack-plugin": "^3.2.0",
    "mini-css-extract-plugin": "^2.3.0",
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^6.0.2",
    "regenerator-runtime": "^0.13.9",
    "resolve-url-loader": "^4.0.0",
    "trix": "^1.2.0",
    "webpack": "^5.64.4",
    "webpack-cli": "^4.8.0"
  },
  "devDependencies": {
    "@babel/eslint-parser": "^7.16.0",
    "@babel/eslint-plugin": "7.14.5",
    "@webpack-cli/serve": "^1.5.2",
    "webpack-dev-server": "^4.6.0"
  },
  "private": true,
  "scripts": {
    "dev": "./bin/webpack-dev-server"
  },
  "version": "0.0.0"
}
```

--- a/actiontext/package.json
+++ b/actiontext/package.json
@@ -2,10 +2,10 @@
   "name": "@rails/actiontext",
   "version": "7.1.0-alpha",
   "description": "Edit and display rich text in Rails applications",
-  "main": "app/javascript/actiontext/index.js",
+  "main": "app/assets/javascripts/actiontext.js",
   "type": "module",
   "files": [
-    "app/javascript/actiontext/*.js"
+    "app/assets/javascripts/*.js"
   ],
   "homepage": "https://rubyonrails.org/",
   "repository": {

